lib/elixir/lib/list.ex

```diff
@@ -591,7 +591,12 @@ defmodule List do
   """
   @spec to_string(:unicode.charlist) :: String.t
   def to_string(list) when is_list(list) do
-    case :unicode.characters_to_binary(list) do
+    try do
+       :unicode.characters_to_binary(list)
+    rescue
+      ArgumentError ->
+        raise ArgumentError, "cannot make this list into a string (use 'inspect(list)' for its internal representation as a string)"
```

Comment:
Something like this, you mean?

``` elixir
 "cannot make #{inspect(list)} into a string (use 'inspect(list)' for its internal representation as a string)"
```

It could make for long (and thus maybe hard to read) errors given long lists, and truncating lists might be going overboard. What do you think?

On a side note, I don't love the message I came up with, but I can't think of a better one at the moment.